src/main/java/seedu/address/logic/commands/DeleteLinkCommand.java

```diff
@@ -1,77 +1,80 @@
-            + "Example: " + COMMAND_WORD + " 1 "
-            + PREFIX_MODULE_LINK + "coursemology.org";
-
-    public static final String MESSAGE_DELETE_LINK_SUCCESS = "Deleted link from module: %1$s";
+    public static final String MESSAGE_USAGE = "[" + COMMAND_WORD + "]: Delete link/s from a module "
+            + "using its module code and user-defined alias.\n"
+            + "A 'm/' flag should be appended to the front the module code;\n"
+            + "A 'la/' flag should be appended to the front of each link alias.\n"
+            + "Example: " + COMMAND_WORD + " " + PREFIX_MODULE_CODE + "GEA1000 "
+            + PREFIX_MODULE_LINK_ALIAS + "coursemo " + PREFIX_MODULE_LINK_ALIAS + "kattis";
+
+    public static final String MESSAGE_DELETE_LINK_SUCCESS = "Successfully deleted link/s from module code [%1$s]!";
     public static final String MESSAGE_NOT_EDITED = "At least one link must be deleted.";
-    public static final String MESSAGE_MISSING_LINK = "This link does not currently exist in module index [";
+    public static final String MESSAGE_MISSING_LINK_ALIAS = "The link alias [%1$s] does not currently exist"
+            + " in the module with module code [%2$s]";
 
-    private final Index index;
-    private final Set<Link> links;
+    private final ModuleCode moduleCode;
+    private final List<String> linkAliases;
 
     /**
      * Creates a DeleteLinkCommand for the deletion of links from a module
-     * @param index index of the module based on the filtered module list
-     * @param links links to delete from the module
+     * @param moduleCode module code of the module in which links will be deleted
+     * @param linkAliases contain aliases of the links to delete from the specified module
      */
-    public DeleteLinkCommand(Index index, Set<Link> links) {
-        requireAllNonNull(index, links);
-        this.index = index;
-        this.links = links;
+    public DeleteLinkCommand(ModuleCode moduleCode, List<String> linkAliases) {
+        requireAllNonNull(moduleCode, linkAliases);
+        this.moduleCode = moduleCode;
+        this.linkAliases = linkAliases;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        ObservableList<Module> lastShownList = model.getFilteredModuleList();
-        if (index.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_MODULE_DISPLAYED_INDEX);
-        }
 
-        Module moduleToEdit = lastShownList.get(index.getZeroBased());
-        Module editedModule = createEditedModule(moduleToEdit, links, index);
+        Module moduleToEdit = null;
+        try {
+            moduleToEdit =
+                    model.getModuleUsingModuleCode(moduleCode, true);
+        } catch (ModuleNotFoundException e) {
+            throw new CommandException(String.format(MESSAGE_NO_MODULE_IN_FILTERED_LIST,
```

Comment:
```suggestion
            throw new CommandException(String.format(Messages.MESSAGE_NO_MODULE_IN_FILTERED_LIST,
```
same as aforementioned comment